### PR TITLE
Use message type instead of prototype

### PR DESCRIPTION
**Pull request by 0x26res**
I've fixed the issue and added some tests. 

But I wasn't reproduce the issue with this repo set up. The issue stems from the fact that in some environment:
```
MessageFactory().GetPrototype(message_type.DESCRIPTOR) is not messsage_type
```

I think it must have something to do with a different combination of `protoc` and `protobuf` python library.

Does anyone happen to know what version of `protoc` has been used to generate the pb2.py files?

Thanks.

### CODE REVIEW

File: tests/integration/schema_registry/test_proto_serializers.py [hunk 1]
```diff
@@ -23,30 +23,10 @@
     PublicTestProto_pb2
 from tests.integration.schema_registry.data.proto.DependencyTestProto_pb2 import DependencyMessage
 from tests.integration.schema_registry.data.proto.exampleProtoCriteo_pb2 import ClickCas
+from tests.schema_registry.test_proto import TEST_DATA
```
Comment (pranavrth) [inline tests/integration/schema_registry/test_proto_serializers.py:26]: It's better to keep data in their respective test files. Its fine here to replicate data in each file if we require that.

File: tests/schema_registry/test_proto.py [hunk 2]
```diff
@@ -19,13 +19,41 @@
...
+from confluent_kafka.serialization import SerializationContext, MessageField
 from tests.integration.schema_registry.data.proto import (DependencyTestProto_pb2,
-                                                          metadata_proto_pb2)
+                                                          metadata_proto_pb2,
+                                                          TestProto_pb2,
+                                                          PublicTestProto_pb2,
+                                                          NestedTestProto_pb2)
+from tests.schema_registry.test_api_client import TEST_URL
 
+TEST_DATA = [
```
Comment (pranavrth) [inline tests/schema_registry/test_proto.py:35]: No need for such complex data while testing our usecase. Keep only what is really required.

Move data near the testcase.

File: tests/integration/schema_registry/test_proto_serializers.py [hunk 3]
```diff
@@ -117,7 +97,6 @@ def test_protobuf_serializer_type_mismatch(kafka_cluster):
                              r" 'tests.integration.schema_registry.data.proto.NestedTestProto_pb2.NestedMessage'\>"):
         producer.produce(topic, key=pb2_2())
 
-
```
Comment (pranavrth) [inline tests/integration/schema_registry/test_proto_serializers.py:120]: Keep this extra line. Each test is separated by 2 lines.

File: src/confluent_kafka/schema_registry/protobuf.py [hunk 4]
```diff
@@ -510,7 +510,8 @@ def __init__(self, message_type, conf=None):
 
         descriptor = message_type.DESCRIPTOR
         self._index_array = _create_index_array(descriptor)
-        self._msg_class = MessageFactory().GetPrototype(descriptor)
+        self._msg_class = message_type
+
```
Comment (pranavrth) [inline src/confluent_kafka/schema_registry/protobuf.py:514]: Remove this extra line.

File: tests/schema_registry/test_proto.py [hunk 5]
```diff
@@ -83,3 +111,27 @@ def test_index_encoder(msg_idx, zigzag, expected_hex):
     buf.seek(0)
     decoded_msg_idx = ProtobufDeserializer._decode_index(buf, zigzag=zigzag)
     assert decoded_msg_idx == msg_idx
+
+
+@pytest.mark.parametrize("pb2, data", TEST_DATA)
+def test_round_trip(mock_schema_registry, pb2, data):
```
Comment (pranavrth) [inline tests/schema_registry/test_proto.py:117]: Please write a better name for the test which suggests what we are trying to test. Something in the line of serializing, deserializing and then again serializing. Better to have verbose test name.

File: tests/schema_registry/test_proto.py [hunk 6]
```diff
@@ -83,3 +111,27 @@ def test_index_encoder(msg_idx, zigzag, expected_hex):
...
+    test_client = mock_schema_registry({"url": TEST_URL})
+    context = SerializationContext("topic-name", MessageField.VALUE)
+    serializer = ProtobufSerializer(pb2, test_client, conf=conf)
+    deserializer = ProtobufDeserializer(pb2, conf=conf)
+
+    message_before = pb2(**data)
+    bytes_before = serializer(message_before, context)
+    message_after = deserializer(bytes_before, context)
+    bytes_after = serializer(message_after, context)
+    assert isinstance(message_after, pb2)
```
Comment (pranavrth) [inline tests/schema_registry/test_proto.py:129]: Keep all asserts continuously.

Add a line break before assert.

File: tests/schema_registry/test_proto.py [hunk 7]
```diff
@@ -83,3 +111,27 @@ def test_index_encoder(msg_idx, zigzag, expected_hex):
...
+
+    assert message_before == message_after
+    assert bytes_after == bytes_before
+
+
+def test_foo_bar():
```
Comment (pranavrth) [inline tests/schema_registry/test_proto.py:135]: We are removing this code altogether, so I think we don't need to test this. Please remove this function.

**Review by pranavrth — CHANGES_REQUESTED** [review]
Please fix the build issues. Mostly related to the styling. 

https://app.travis-ci.com/github/confluentinc/confluent-kafka-python/builds/254834885#L827

**Comment by 0x26res** [comment]
@pranavrth thanks for the feed back. I've addressed the comments.

The CI is failing, but (unless I'm mistaken) I think the serdes branch has been left in a bad state, some tests need fixing.

**Comment by pranavrth** [comment]
Hello @0x26res,

The testcases related to protobuf were failing in **serdes** branch. I have fixed and merged the test cases. Please resolve the conflict and it should work fine.

**Comment by 0x26res** [comment]
@pranavrth thanks for fixing, it looks good now.

**Comment by pranavrth** [comment]
LGTM

**Review by pranavrth — APPROVED** [review]

**Comment by 0x26res** [comment]
@pranavrth I don't have write access so I won't be able to merge. Can I leave this to you? Thanks

**Comment by 0x26res** [comment]
@mhowlett is that not going through? Should I send another MR?

**Pull request closed without merging**